src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private User user;
+    private Role userRole;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {User.class, Role.class, Role.RoleName.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        userRole = new Role(Role.RoleName.USER);
+        user = new User();
+        user.setEmail("bob@i.ua");
+        user.setPassword("1234");
+    }
+
+    @Test
+    void save_Ok() {
+        User actual = userDao.save(user);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(user, actual);
```

Comment:
User.class don`t have overrided equals method, so in this case, you compare the reference to the object.
You can compare the fields of the object